nixos/modules/services/misc/anki-sync-server.nix

```diff
@@ -0,0 +1,153 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+with lib; let
+  cfg = config.services.anki-sync-server;
+  name = "anki-sync-server";
+  specEscape = replaceStrings ["%"] ["%%"];
+  usersWithIndexes =
+    lists.imap1 (i: user: {
+      i = i;
+      user = user;
+    })
+    cfg.users;
+  usersWithIndexesFile = filter (x: x.user.passwordFile != null) usersWithIndexes;
+  usersWithIndexesNoFile = filter (x: x.user.passwordFile == null && x.user.password != null) usersWithIndexes;
+  anki-sync-server-run = pkgs.writeShellScriptBin "anki-sync-server-run" ''
+    # When services.anki-sync-server.users.passwordFile is set,
+    # each password file is passed as a systemd credential, which is mounted in
+    # a file system exposed to the service. Here we read the passwords from
+    # the credential files to pass them as environment variables to the Anki
+    # sync server.
+    ${
+      concatMapStringsSep
+      "\n"
+      (x: ''export SYNC_USER${toString x.i}=${strings.escapeShellArg x.user.username}:"''$(cat "''${CREDENTIALS_DIRECTORY}/"${strings.escapeShellArg x.user.username})"'')
+      usersWithIndexesFile
+    }
+    # For users where services.anki-sync-server.users.password isn't set,
+    # export passwords in environment variables in plaintext.
+    ${
+      concatMapStringsSep
+      "\n"
+      (x: ''export SYNC_USER${toString x.i}=${strings.escapeShellArg x.user.username}:${strings.escapeShellArg x.user.password}'')
+      usersWithIndexesNoFile
+    }
+    exec ${cfg.package}/bin/anki --syncserver
```

Comment:
Would be great if someone would split the built-in server:
https://docs.ankiweb.net/sync-server.html#with-cargo

That should generate a much smaller package (without qt etc dependencies)

Doesn't have to be done in this PR, but that was one of the reasons I didn't do this back when I added anki-sync-server-rs (the standalone server didn't exist at the time)

Reply:
Noted

Reply:
Done in https://github.com/NixOS/nixpkgs/pull/261166 (Cc'd you directly)

Reply:
Let's decide whether to merge this one or the other.

Reply:
I think it's easier to deal with small PRs (that one, a PR to update anki to 23.10, then this) but either work for me; I wanted to try to get this in small bits before nixos 23.11 but I think we're a wee late now so either work for me :)

For this PR as of now, we can't take https://github.com/NixOS/nixpkgs/pull/257692/commits/d7ae7c637c487147f83019752864e0a29fa8bbe7 (update anki to 23.10beta6)  -- I didn't keep updating the old branch but we'll need it to be a stable version (request from anki maintainers in the other PR, but it totally makes sense); anki 23.10.1 got released since so we can go ahead and upgrade to that.
Do you want to give it a try or shall I?

As for which PR to merge, I really don't mind either way; if this is ready I'm happy to close the other one. Otherwise I'll post it to the discourse topic of PRs ready to be merged and it shouldn't take too long...

Reply:
Haven't done anything for anki 23.10.1, but I just did 23.10beta6 -> 23.10.1 for anki-sync-server-rs's anki dependency as a last struggle without any trouble so I expect it'll be fine just updating hashes (src and cargo deps).
I'll let you update this and close the other PR

Reply:
I updated to Anki 23.10.1 in this PR.

Reply:
I updated to 23.10.1